EventFilter/SiPixelRawToDigi/interface/GPUSimpleVector.hpp

```diff
@@ -0,0 +1,72 @@
+//  author: Felice Pantaleo, CERN, 2018
+#ifndef GPU_SIMPLEVECTOR_HPP_
+#define GPU_SIMPLEVECTOR_HPP_
+
+namespace GPU {
+template <class T> struct SimpleVector {
+  // Constructors
+  __host__ __device__ SimpleVector(unsigned int maxSize, T *m_data = nullptr)
+      : m_size(0), m_data(m_data), maxSize(static_cast<int>(maxSize)) {}
+
+  __host__ __device__ SimpleVector() : SimpleVector(0) {}
+
+  __inline__ __host__ __device__ int push_back(const T &element) {
+
+    auto previousSize = m_size;
+    m_size++;
+    if (previousSize < maxSize) {
+      m_data[previousSize] = element;
+      return previousSize;
+    } else {
+      --m_size;
+      return -1;
```

Comment:
More general question on the caller-side error handling (not necessary to cover fully here). I guess it is up to the caller to check the return value, and act accordingly if it gets `-1` (possibly propagating to the host to throw an exception).